### PR TITLE
Rename RestEasy Reactive MP Client to quarkus-rest-client-reactive

### PR DESCRIPTION
**Pull request by michalszynkiewicz**

### CODE REVIEW

**Comment by famod** [comment]
Failed build says:
> Dependencies to extension artifacts are outdated! Run ./update-extension-dependencies.sh and add the modified pom.xml files to your commit.

File: devtools/bom-descriptor-json/pom.xml [hunk 1]
```diff
@@ -1752,7 +1752,7 @@
                 </dependency>
                 <dependency>
                     <groupId>io.quarkus</groupId>
-                    <artifactId>quarkus-rest-client-microprofile</artifactId>
+                    <artifactId>quarkus-rest-client-mutiny</artifactId>
```
Comment (geoand) [inline devtools/bom-descriptor-json/pom.xml:1755]: Are you sure this is correct? If so, why?

Comment (geoand) [reply devtools/bom-descriptor-json/pom.xml:1755]: Basically I have the same question in every `mutiny` related change :)

Comment (michalszynkiewicz) [reply devtools/bom-descriptor-json/pom.xml:1755]: what happened here :D thanks! fixing

Comment (michalszynkiewicz) [reply devtools/bom-descriptor-json/pom.xml:1755]: oh, it's the dependencies something script. It changed the order of dependencies.

For a moment I didn't know what's going on :)

Comment (michalszynkiewicz) [reply devtools/bom-descriptor-json/pom.xml:1755]: The realc hange is `quarkus-rest-client-microprofile` -> `quarkus-rest-client-reactive`

Comment (geoand) [reply devtools/bom-descriptor-json/pom.xml:1755]: That's what I though :)

Comment (geoand) [reply devtools/bom-descriptor-json/pom.xml:1755]: Thanks for figuring it out, it baffled me :)

**Review by geoand — APPROVED** [review]
LGTM

File: devtools/bom-descriptor-json/pom.xml (hunk 1 above)
Comment (famod) [reply devtools/bom-descriptor-json/pom.xml:1755]: > oh, it's the dependencies something script

It's magic! 😆 
Someone wanted it to be sorted alphabetically, which makes sense, but also leads to something like this...

**Pull request merged**